src/tasks/select-component.ts

```diff
@@ -0,0 +1,85 @@
+import inquirer from "inquirer";
+import inquirerSearchCheckbox from "inquirer-search-checkbox";
+import { ResourceContext, ZeplinComponent } from "./context/resource";
+import { TaskStep, Task, transitionTo, pauseSpinningAndExecuteTask } from "../util/task";
+import { ComponentSection, Component } from "../api/interfaces";
+import { ZeplinApi } from "../api";
+import * as ui from "./ui/select-component";
+
+inquirer.registerPrompt("search-checkbox", inquirerSearchCheckbox);
+
+const zeplinApi = new ZeplinApi();
+
+function mapComponents(components?: Component[]): ZeplinComponent[] {
+    return (components || []).map(c => ({
+        _id: c._id,
+        name: c.name,
+        sourceId: c.sourceId
+    }));
+}
+
+function extractComponents(componentSections?: ComponentSection[]): ZeplinComponent[] {
+    const components: ZeplinComponent[] = [];
+
+    (componentSections || []).forEach(cs => {
```

Comment:
```suggestion
function extractComponents(componentSections: ComponentSection[] = []): ZeplinComponent[] {
    const components: ZeplinComponent[] = [];

    componentSections.forEach(cs => {
```
BTW the current version is also acceptable.

Reply:
In addition to the suggestion, I think `reduce` function is better than `forEach` regarding the code here. Since, using `reduce` or `forEach` is a bit preference, this comment is not a suggestion. 😄 Just wanted to mention.